llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1939,10 +1943,7 @@ void HWAddressSanitizer::ShadowMapping::init(Triple &TargetTriple,
   } else if (ClWithIfunc) {
     Kind = OffsetKind::kIfunc;
     WithFrameRecord = false;
-  } else if (ClWithTls) {
```

Comment:
I don't understand why we flipped this logic. before all the ifs were positive

Reply:
ClWithTls branch is now empty - everything is in default

Reply:
I understand, but standing on it's own, it's unclear why this CL makes things better.

Reply:
It's NFC which does not need make things better, just not make things worse. The point is to instruct this shuffling to have smaler non-NFC patch to get to https://github.com/llvm/llvm-project/pull/109621/files.

Reply:
Added description if it helps

Reply:
yes